web/src/main/java/com/bazaarvoice/emodb/web/migrator/LocalRangeMigrator.java

```diff
@@ -34,27 +35,29 @@
     private final Meter _failedRangeMigrations;
 
     private final MigratorTools _migratorTools;
-    private final MigratorRateLimiter _rateLimiter;
+    private final MigratorRateLimiter _migratorRateLimiter;
 
     @Inject
     public LocalRangeMigrator(MigratorTools migratorTools, LifeCycleRegistry lifecycle, MetricRegistry metricRegistry,
                               MigratorRateLimiter rateLimiter) {
         _migratorTools = migratorTools;
-        _rateLimiter = rateLimiter;
+        _migratorRateLimiter = rateLimiter;
 
         _activeRangeMigrations = metricRegistry.counter(MetricRegistry.name("bv.emodb.migrator", "Migrator", "active-range-migrations"));
         _failedRangeMigrations = metricRegistry.meter(MetricRegistry.name("bv.emodb.migrator", "Migrator", "failed-range-migrations"));
     }
 
     public RangeScanUploaderResult migrate(final int taskId, ScanOptions options, final String placement, ScanRange range,
                                            String migrationId) throws IOException, InterruptedException {
-        
+
         _log.info("Migrating placement {}: {}", placement, range);
 
         Supplier<Integer> writeRate = Suppliers.memoizeWithExpiration(
-                () -> _rateLimiter.getMaxWritesPerSecond(migrationId),
+                () -> _migratorRateLimiter.getMaxWritesPerSecond(migrationId),
                 READ_RATE_LIMITER_SECONDS, TimeUnit.SECONDS);
 
+        RateLimiter rateLimiter = RateLimiter.create(writeRate.get());
```

Comment:
Huh, I didn't know you could modify the rate of an existing rate limiter. Learned something.